### PR TITLE
[SYCL] Fix device assertion bug on windows

### PR DESCRIPTION
**Pull request by lbushi25**
This PR fixes a device assertion bug on Windows where the assertion message and information are not printed when an assertion is triggered. Also re-enabled some tests failing because of it. See https://github.com/intel/llvm/issues/12797.

### CODE REVIEW

File: sycl/include/sycl/stl_wrappers/assert.h [hunk 1]
```diff
@@ -0,0 +1,45 @@
+//==---------------- <assert.h> wrapper around STL
+//--------------------------==//
```
Comment (aelovikov-intel) [inline sycl/include/sycl/stl_wrappers/assert.h:2]: bad formatting

**Review by aelovikov-intel — APPROVED** [review]
Please fix formatting, other than that LGTM.

**Comment by lbushi25** [comment]
> Please fix formatting, other than that LGTM.

Done!

**Comment by lbushi25** [comment]
@intel/llvm-gatekeepers 
This is ready for merge. All failures are infrastructure related in the cleanup stage.

**Comment by lbushi25** [comment]
@intel/llvm-gatekeepers ping

**Comment by aelovikov-intel** [comment]
> @intel/llvm-gatekeepers ping

Please fix:

image

**Comment by lbushi25** [comment]
> > @intel/llvm-gatekeepers ping
> 
> Please fix:
> 
> image

I haven't seen this before. Where is it from and how can I fix it?

**Comment by lbushi25** [comment]
Reminder to close  #12797 once this is merged.

**Comment by steffenlarsen** [comment]
> > > @intel/llvm-gatekeepers ping
> > 
> > 
> > Please fix:
> > image
> 
> I haven't seen this before. Where is it from and how can I fix it?

https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address mentions a "noreply" github email. You may have to edit some of your github settings.

**Comment by lbushi25** [comment]
Opened another clone here with hopefully proper email this time: https://github.com/intel/llvm/pull/15232.
Closing this.

**Pull request closed without merging**